// Copyright 2015-2016 gRPC authors.

syntax = "proto3";

package grpc.testing;

message GaugeResponse {
  string name = 1;
  oneof value {
    int64 long_value = 2;
    double double_value = 3;
    string string_value = 4;
  }
}

message GaugeRequest {
  string name = 1;
}

message EmptyMessage {}

service MetricsService {

  rpc GetAllGauges(EmptyMessage) returns (stream GaugeResponse);

  rpc GetGauge(GaugeRequest) returns (GaugeResponse);
}
